warp10/src/main/java/io/warp10/script/functions/HHCODEFUNC.java

```diff
@@ -0,0 +1,138 @@
+//
+//   Copyright 2020  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.functions;
+
+import com.geoxp.geo.HHCodeHelper;
+import com.google.common.primitives.Longs;
+
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+
+/**
+ * Template function to interface with HHCodeHelper
+ */
+public class HHCODEFUNC extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+
+  public enum HHCodeAction {
+    NORTH, SOUTH, EAST, WEST, NORTH_EAST, NORTH_WEST, SOUTH_EAST, SOUTH_WEST, BBOX, CENTER
+  }
+
+  private HHCodeAction action;
+  private boolean isLongFormat;
```

Comment:
This should not be a field, it should be a local variable updated for each invocation of the function